src/libraries/System.Security.Cryptography.Xml/tests/SignedXmlTest.cs

```diff
@@ -1677,36 +1676,80 @@ public void VerifyXmlResolver(bool provideResolver)
                 catch
                 {
                 }
-
-                ((IDisposable)listener).Dispose();
             }
 
             static async Task ProcessRequests(
-                HttpListener listener,
-                Action<HttpListenerRequest> requestReceived,
+                TcpListener listener,
+                Action requestReceived,
                 CancellationToken cancellationToken)
             {
+                const string Response =
+                    "HTTP/1.1 200 OK\r\nContent-Type: text/plain; charset=UTF-8\r\nContent-Length: 0\r\n\r\n";
+
                 while (!cancellationToken.IsCancellationRequested)
                 {
-                    HttpListenerContext ctx;
+                    Socket socket;
 
                     try
                     {
-                        ctx = await listener.GetContextAsync();
+#if NETCOREAPP
+                        socket = await listener.AcceptSocketAsync(cancellationToken);
+#else
+                        socket = await listener.AcceptSocketAsync();
+#endif
                     }
                     catch
                     {
                         break;
                     }
 
-                    HttpListenerRequest req = ctx.Request;
-                    requestReceived(req);
-
-                    using (HttpListenerResponse resp = ctx.Response)
+                    using (socket)
                     {
-                        resp.ContentType = "text/plain";
-                        resp.ContentEncoding = Encoding.UTF8;
-                        resp.ContentLength64 = 0;
+                        requestReceived();
+                        byte[] buf = new byte[1024];
+
+#if NETCOREAPP
+                        // The CoreFX XML libraries throw an exception if we don't drain the request
+                        // (unless we don't dispose the socket).
+                        // For some reason each ReceiveAsync only gets 10 bytes at a time,
```

Comment:
Probably https://github.com/dotnet/runtime/blob/3968ceb4297d87109ea32401da02fe4432188a4a/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L23-L28

The reads are not guarantied to come as written so the loop is probably right anyway.